src/codegeneration/CommonJsModuleTransformer.js

```diff
@@ -83,7 +83,7 @@ export class CommonJsModuleTransformer extends ModuleTransformer {
     // "module.exports", so we don't append "module.exports = {}" to the output.
     if (this.hasExports()) {
       var descriptors = this.transformObjectLiteralToDescriptors(exportObject);
-      var exportStatement = parseStatement `Object.defineProperties(exports, ${descriptors});`;
+      var exportStatement = parseStatement `Object.defineProperties(module.exports, ${descriptors});`;
       statements = prependStatements(statements, exportStatement);
     }
```

Comment:
If `module.exports` is not created then the downstream detection will fail (according to the motivation for this PR). But the `module.exports` won't be created in the `else` branch, so it seems like all cases are not fully covered.   
But this does not make the current PR invalid just -- maybe --  not enough. More likely @guybedford will explain.

LGTM